daq/host.py

```diff
@@ -325,12 +324,22 @@ def terminate(self, reason, trigger=True):
                     self._host_name(), trigger, reason)
         self._release_config()
         self._state_transition(_STATE.TERM)
-        self.record_result(self.test_name, state=MODE.TERM)
         self._monitor_cleanup()
         self.runner.network.delete_mirror_interface(self.target_port)
+        self._report.finalize()
+        json_path = self._report.path + ".json"
+        with open(json_path, 'w') as json_file:
+            json.dump(self._report.get_all_results(), json_file)
+        remote_paths = {}
+        upload_path = self._get_unique_upload_path(self._report.path)
```

Comment:
I would create a helper function here... I'm seeing 3x uses of the exact same code pattern